contracts/HubPool.sol

```diff
@@ -549,64 +549,70 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
      * relay roots to the SpokePool on the network specified in the leaf.
      * @dev In some cases, will instruct spokePool to send funds back to L1.
      * @notice Deletes the published root bundle if this is the last leaf to be executed in the root bundle.
-     * @param poolRebalanceLeaf Contains all data neccessary to reconstruct leaf contained in root bundle and to
-     * bridge tokens to HubPool. This data structure is explained in detail in the HubPoolInterface.
+     * @param chainId ChainId number of the target spoke pool on which the bundle is executed.
+     * @param bundleLpFees Array representing the total LP fee amount per token in this bundle for all bundled relays.
+     * @param netSendAmounts Array representing the amount of tokens to send to the SpokePool on the target chainId.
+     * @param runningBalances Array used to track any unsent tokens that are not included in the netSendAmounts.
+     * @param leafId Index of this executed leaf within the poolRebalance tree.
+     * @param l1Tokens Array of all the tokens associated with the bundleLpFees, nedSendAmounts and runningBalances.
      * @param proof Inclusion proof for this leaf in pool rebalance root in root bundle.
      */
-    function executeRootBundle(PoolRebalanceLeaf memory poolRebalanceLeaf, bytes32[] memory proof) public nonReentrant {
+    function executeRootBundle(
+        uint256 chainId,
+        uint256[] memory bundleLpFees,
+        int256[] memory netSendAmounts,
+        int256[] memory runningBalances,
+        uint8 leafId,
+        address[] memory l1Tokens,
+        bytes32[] memory proof
+    ) public nonReentrant {
         require(getCurrentTime() > rootBundleProposal.requestExpirationTimestamp, "Not passed liveness");
 
         // Verify the leafId in the poolRebalanceLeaf has not yet been claimed.
-        require(!MerkleLib.isClaimed1D(rootBundleProposal.claimedBitMap, poolRebalanceLeaf.leafId), "Already claimed");
+        require(!MerkleLib.isClaimed1D(rootBundleProposal.claimedBitMap, leafId), "Already claimed");
 
         // Verify the props provided generate a leaf that, along with the proof, are included in the merkle root.
         require(
-            MerkleLib.verifyPoolRebalance(rootBundleProposal.poolRebalanceRoot, poolRebalanceLeaf, proof),
+            MerkleLib.verifyPoolRebalance(
+                rootBundleProposal.poolRebalanceRoot,
+                PoolRebalanceLeaf({
```

Comment:
Just OOC, how does the gas of this method compare (I think we have gas tests for execution now)? I worry that with large arrays, we may end up incurring significant gas costs when copying. Also not sure if structs can contain references to memory or have to copy all their constituents.